bleak/backends/bluezdbus/discovery.py

```diff
@@ -53,50 +53,178 @@ def _device_info(path, props):
         return None, None, None, None
 
 
-async def discover(timeout=5.0, loop=None, **kwargs):
-    """Discover nearby Bluetooth Low Energy devices.
+def _parse_device(path, props):
+    if not props:
+        logger.debug(
+            "Disregarding %s since no properties could be obtained." % path
+        )
+        return
+
+    name, address, _, path = _device_info(path, props)
+    if address is None:
+        return
+
+    uuids = props.get("UUIDs", [])
+    manufacturer_data = props.get("ManufacturerData", {})
+    return BLEDevice(
+            address,
+            name,
+            {"path": path, "props": props},
+            uuids=uuids,
+            manufacturer_data=manufacturer_data,
+        )
 
-    For possible values for `filter`, see the parameters to the
-    ``SetDiscoveryFilter`` method in the `BlueZ docs
-    <https://git.kernel.org/pub/scm/bluetooth/bluez.git/tree/doc/adapter-api.txt?h=5.48&id=0d1e3b9c5754022c779da129025d493a198d49cf>`_
 
-    The ``Transport`` parameter is always set to ``le`` by default in Bleak.
+class AsyncDiscovery():
 
-    Args:
-        timeout (float): Duration to scan for.
-        loop (asyncio.AbstractEventLoop): Optional event loop to use.
+    def __init__(self, callback, loop=None, device="hci0"):
```

Comment:
Maybe make the `callback` optional and add explicit type decorations?
```suggestion
    def __init__(self, callback: Callable[[bleak.BLEDevice], None] = None, loop: asyncio.AbstractEventLoop = None, device="hci0"):
```

Reply:
ah, I thought the bleak guy wants the types in the docstrings. But you are right, they are explicit in other places as well.

Reply:
oh but i screwed up my suggestion and fixed it :sweat_smile:

Reply:
why the Union? Do you have a link?

Reply:
https://docs.python.org/3/library/typing.html#typing.Union
But i just have seen that there is `Optional[X]` which seems a much nicer solution.
Thanks for making me read the docs again :+1: